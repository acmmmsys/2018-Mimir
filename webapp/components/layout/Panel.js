import React, { Component } from 'react';
import classnames from 'classnames';
import { compose, withProps, branch, nest, renderComponent, defaultProps } from 'recompose';
import { Grid, Loader } from 'semantic-ui-react';

import { isEmpty } from 'utils';

import 'style/layout/Panel.scss';

const PanelLoading = props =>
    <Grid.Row className='panel'>
        <div style={{position: 'relative', margin: '0 auto'}}>
            <Loader active />
        </div>
    </Grid.Row>

const PanelEmpty = props =>
    <Grid.Row className='panel'>
        {props.emptyMessage}
    </Grid.Row>

class Panel extends Component {
    render()Â {
        let classNames = classnames([
            'panel'
        ]);

        return (
            <Grid.Row className={classNames}>
                {this.props.children}
            </Grid.Row>
        );
    }
}

const enhance = compose(
    defaultProps({ 
        emptyMessage: '',
        loading: false,
    }),
    branch(
        props => props.loading,
        renderComponent(PanelLoading)
    ),
    branch(
        props => isEmpty(props.children) || props.isEmpty,
        renderComponent(PanelEmpty)
    )
);

export default enhance(Panel);