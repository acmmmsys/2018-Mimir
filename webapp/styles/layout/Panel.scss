@import '../colors';

.panel {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-wrap: wrap;

    max-width: 100%;
    max-height: 100%;

    border-left: 2px solid lighten($backgroundColor, 5%);
    border-right: 2px solid darken($backgroundColor, 5%);
    border-top: 2px solid lighten($backgroundColor, 5%);
    border-bottom: 2px solid darken($backgroundColor, 5%);
}

.selection-panel {
    overflow-y: auto;
}