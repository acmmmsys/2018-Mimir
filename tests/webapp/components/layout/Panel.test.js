import React from 'react';
import Panel from '../../../../webapp/components/layout/Panel'
import renderer from 'react-test-renderer';

let wrapper;

test('Panel does render', () => {
    wrapper = shallow(
        <Panel>

        </Panel>
    );
    expect(wrapper).toMatchSnapshot();
});
