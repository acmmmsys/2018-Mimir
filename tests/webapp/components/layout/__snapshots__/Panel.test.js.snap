// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel does render 1`] = `
<branch(branch(Panel))
  emptyMessage=""
  loading={false}
/>
`;
